src/jgikbase/test/idmapping/cli_test.py

```diff
@@ -0,0 +1,131 @@
+from jgikbase.idmapping.builder import IDMappingBuilder, IDMappingBuildException
+from unittest.mock import create_autospec, Mock
+from jgikbase.idmapping.cli import IDMappingCLI
+from jgikbase.idmapping.core.user_handler import LocalUserHandler
+from jgikbase.idmapping.core.user import Username
+from pathlib import Path
+from pytest import raises
+from jgikbase.test.idmapping.test_utils import assert_exception_correct
+
+# TODO CLI at some point, test usage and invalid args. Since argparse calls exit() when this
+# happens, it'll need exec() tests, or futzing with argparse.
+
+
+def test_init_fail_None_input():
+    b = create_autospec(IDMappingBuilder, spec_set=True, instance=True)
+    out = Mock()
+    err = Mock()
+
+    fail_init(None, [], out, err, TypeError('builder cannot be None'))
+    fail_init(b, None, out, err, TypeError('args cannot be None'))
+    fail_init(b, ['-h', None], out, err, TypeError('None item in args'))
+    fail_init(b, [], None, err, TypeError('stdout cannot be None'))
+    fail_init(b, [], out, None, TypeError('stderr cannot be None'))
+
+
+def fail_init(builder, args, out, err, expected):
+    with raises(Exception) as got:
+        IDMappingCLI(builder, args, out, err)
+    assert_exception_correct(got.value, expected)
+
+
+def test_no_input():
+    builder = create_autospec(IDMappingBuilder, spec_set=True, instance=True)
+    out = Mock()
+    err = Mock()
+
+    assert IDMappingCLI(builder, [], out, err).execute() == 1
+
+    assert out.write.call_args_list == []
+    assert err.write.call_args_list == [
+        (('One of --list-users or --user must be specified.\n',), {})]
+
+
+def test_fail_build():
```

Comment:
This and the next test should throw the exception on build_local_user_handler, not get_users. Fixed locally.